src/plugins/simulator/model/simulator-content.ts

```diff
@@ -114,8 +138,15 @@ export const SimulatorContentModel = TileContentModel
           sharedModelManager.addTileSharedModel(self, containerSharedModel);
         }
 
-        if(sharedProgramModel && !tileSharedModels?.includes(sharedProgramModel)) {
-          sharedModelManager.addTileSharedModel(self, sharedProgramModel);
+        if (programsWithDataflowTiles.length > 0) {
```

Comment:
I tested it and it works as it looks like it should:

If  we are attached to a program and that program is not on the list of programs with tiles, we should
     - unattach
     - attach to the first one on the list of programs that do have tiles

Otherwise we should attach to the first program with a tile